### ISSUE TITLE
RTP servers turning into zombies

### ISSUE THREAD

**Issue by pkgw** [HERA-Team/RTP#16]
We're observing a failure mode where the RTP servers on the still nodes can turn into zombies. The servers are still running, but they stop getting assigned new jobs, leaving their machines idle and reducing our throughput.

The problem is that the RTP server has to "check in" with the client (the manager running on `qmaster`) periodically. The periodic checkin is performed on a separate Python thread, and this thread is crashing. The main thread keeps on running, though, so it's not as obvious that the server is incapacitated.

I've seen the same backtrace every time this happens:

```
Exception in thread Thread-3:                         
Traceback (most recent call last):
  File "/home/obs/anaconda/envs/HERA/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/home/obs/anaconda/envs/HERA/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/obs/src/RTP/lib/task_server.py", line 618, in checkin_timer
    ip_addr = socket.gethostbyname(hostname)
gaierror: [Errno -2] Name or service not known
```

This basically means that the server is unable to look up its own IP address, for whatever reason. This particular event has a timestamp of 2017-09-18 16:57:20 if one wants to go spelunking in logs for an explanation. It appears to be a transient problem since the servers start running just fine, and can be restarted just fine as well.

One could save the results if the `ip_addr` lookup, wrap the relevant statements in a `try`/`catch`, and reuse the previous results if they fail. It would be nice to understand why exactly these calls were failing, though. Some debugging `print` statements might be helpful.

Another approach might be to reuse [the code in hera_mc](https://github.com/HERA-Team/hera_mc/blob/master/scripts/mc_server_status_daemon.py#L39-L60) that does this lookup, which tries to look up the machine's IP address in the officially-sanctioned way. Depending on the cause of the problem, that code might crash too.

**Comment by plaplant** [comment]
A fix was attempted in PR #17, but I'll wait to close this issue until we're reasonably convinced that it's gone away for good.

**Comment by pkgw** [comment]
Fixed.
